### ISSUE TITLE
infinite update loop with conda update -n base -c defaults --all

### ISSUE THREAD

**Issue by settle** [conda/conda#8765]
## Current Behavior
Conda updates, but gets in a state that it ping-pongs back and forth between versions to upgrade and downgrade if you immediately repeat update process.

### Steps to Reproduce
```
bash-4.1$ conda update -n base -c defaults --all
Collecting package metadata: done
Solving environment: done

## Package Plan ##

  environment location: /home/<USER>/nobkup/anaconda2


The following packages will be DOWNGRADED:

  cycler                                      0.10.0-py27_0 --> 0.10.0-py27hc7354d3_0
  funcsigs                                     1.0.2-py27_0 --> 1.0.2-py27h83f16ab_0
  ipython_genutils                     0.2.0-py27h89fb69b_0 --> 0.2.0-py27_0
  nbformat                             4.4.0-py27hed7f2b2_0 --> 4.4.0-py27_0
  snowballstemmer                      1.2.1-py27h44e2768_0 --> 1.2.1-py27_0
  unicodecsv                          0.14.1-py27h5062da9_0 --> 0.14.1-py27_0
  wcwidth                                      0.1.7-py27_0 --> 0.1.7-py27h9e3e1ab_0
  xlwt                                 1.3.0-py27h3d85d97_0 --> 1.3.0-py27_0


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
bash-4.1$ conda update -n base -c defaults --all
Collecting package metadata: done
Solving environment: done

## Package Plan ##

  environment location: /home/<USER>/nobkup/anaconda2


The following packages will be DOWNGRADED:

  contextlib2                                  0.5.5-py27_0 --> 0.5.5-py27hbf4c468_0
  docutils                                      0.14-py27_0 --> 0.14-py27hae222c1_0
  ipython_genutils                             0.2.0-py27_0 --> 0.2.0-py27h89fb69b_0
  locket                                       0.2.0-py27_1 --> 0.2.0-py27h73929a2_1
  nbformat                                     4.4.0-py27_0 --> 4.4.0-py27hed7f2b2_0
  prompt_toolkit                              1.0.15-py27_0 --> 1.0.15-py27h1b593e1_0
  singledispatch                     3.4.0.3-py27h9bcb476_0 --> 3.4.0.3-py27_0
  snowballstemmer                              1.2.1-py27_0 --> 1.2.1-py27h44e2768_0
  unicodecsv                                  0.14.1-py27_0 --> 0.14.1-py27h5062da9_0
  wcwidth                              0.1.7-py27h9e3e1ab_0 --> 0.1.7-py27_0
  xlwt                                         1.3.0-py27_0 --> 1.3.0-py27h3d85d97_0


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
bash-4.1$ conda update -n base -c defaults --all
Collecting package metadata: done
Solving environment: done

## Package Plan ##

  environment location: /home/<USER>/nobkup/anaconda2


The following packages will be DOWNGRADED:

  bkcharts                               0.2-py27h241ae91_0 --> 0.2-py27_0
  docutils                              0.14-py27hae222c1_0 --> 0.14-py27_0
  et_xmlfile                                   1.0.1-py27_0 --> 1.0.1-py27h75840f5_0
  funcsigs                             1.0.2-py27h83f16ab_0 --> 1.0.2-py27_0
  locket                               0.2.0-py27h73929a2_1 --> 0.2.0-py27_1
  nbformat                             4.4.0-py27hed7f2b2_0 --> 4.4.0-py27_0
  prompt_toolkit                      1.0.15-py27h1b593e1_0 --> 1.0.15-py27_0
  traitlets                            4.3.2-py27hd6ce930_0 --> 4.3.2-py27_0
  xlwt                                 1.3.0-py27h3d85d97_0 --> 1.3.0-py27_0


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
bash-4.1$ conda update -n base -c defaults --all
Collecting package metadata: done
Solving environment: done

## Package Plan ##

  environment location: /home/<USER>/nobkup/anaconda2


The following packages will be DOWNGRADED:

  bkcharts                                       0.2-py27_0 --> 0.2-py27h241ae91_0
  contextlib2                          0.5.5-py27hbf4c468_0 --> 0.5.5-py27_0
  funcsigs                                     1.0.2-py27_0 --> 1.0.2-py27h83f16ab_0
  ipython_genutils                     0.2.0-py27h89fb69b_0 --> 0.2.0-py27_0
  locket                                       0.2.0-py27_1 --> 0.2.0-py27h73929a2_1
  prompt_toolkit                              1.0.15-py27_0 --> 1.0.15-py27h1b593e1_0
  singledispatch                             3.4.0.3-py27_0 --> 3.4.0.3-py27h9bcb476_0
  traitlets                                    4.3.2-py27_0 --> 4.3.2-py27hd6ce930_0
  wcwidth                                      0.1.7-py27_0 --> 0.1.7-py27h9e3e1ab_0


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
bash-4.1$ conda update -n base -c defaults --all
Collecting package metadata: done
Solving environment: done

## Package Plan ##

  environment location: /home/<USER>/nobkup/anaconda2


The following packages will be DOWNGRADED:

  bkcharts                               0.2-py27h241ae91_0 --> 0.2-py27_0
  cycler                              0.10.0-py27hc7354d3_0 --> 0.10.0-py27_0
  et_xmlfile                           1.0.1-py27h75840f5_0 --> 1.0.1-py27_0
  nbformat                                     4.4.0-py27_0 --> 4.4.0-py27hed7f2b2_0
  prompt_toolkit                      1.0.15-py27h1b593e1_0 --> 1.0.15-py27_0
  unicodecsv                          0.14.1-py27h5062da9_0 --> 0.14.1-py27_0
  wcwidth                              0.1.7-py27h9e3e1ab_0 --> 0.1.7-py27_0


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
```

## Expected Behavior
If you conda update immediately following a successful conda update, then nothing should be further upgraded or downgraded.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : None
            shell level : 0
       user config file : /home/<USER>/.condarc
 populated config files : 
          conda version : 4.6.14
    conda-build version : 3.17.8
         python version : 2.7.16.final.0
       base environment : /home/<USER>/nobkup/anaconda2  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/<USER>/nobkup/anaconda2/pkgs
                          /home/<USER>/.conda/pkgs
       envs directories : /home/<USER>/nobkup/anaconda2/envs
                          /home/<USER>/.conda/envs
               platform : linux-64
             user-agent : conda/4.6.14 requests/2.22.0 CPython/2.7.16 Linux/4.4.0-59-generic ubuntu/14.04.5 glibc/2.19
                UID:GID : 17766:10115
             netrc file : None
           offline mode : False

```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
<NONE>
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at /home/<USER>/nobkup/anaconda2:
#
# Name                    Version                   Build  Channel
_ipyw_jlab_nb_ext_conf    0.1.0                    py27_0    defaults
alabaster                 0.7.12                   py27_0    defaults
anaconda                  custom           py27h4a00acb_0    defaults
anaconda-client           1.7.2                    py27_0    defaults
anaconda-navigator        1.9.7                    py27_0    defaults
anaconda-project          0.8.2                    py27_0    defaults
asn1crypto                0.24.0                   py27_0    defaults
astroid                   1.6.5                    py27_0    defaults
astropy                   2.0.9            py27hdd07704_0    defaults
atomicwrites              1.3.0                    py27_1    defaults
attrs                     19.1.0                   py27_1    defaults
babel                     2.6.0                    py27_0    defaults
backports                 1.0                      py27_1    defaults
backports.functools_lru_cache 1.5                      py27_1    defaults
backports.os              0.1.1                    py27_0    defaults
backports.shutil_get_terminal_size 1.0.0                    py27_2    defaults
backports_abc             0.5                        py_0    defaults
beautifulsoup4            4.7.1                    py27_1    defaults
bitarray                  0.9.3            py27h7b6447c_0    defaults
bkcharts                  0.2                      py27_0    defaults
blas                      1.0                         mkl    defaults
bleach                    3.1.0                    py27_0    defaults
blosc                     1.16.3               hd408876_0    defaults
bokeh                     1.2.0                    py27_0    defaults
boto                      2.49.0                   py27_0    defaults
bottleneck                1.2.1            py27h035aef0_1    defaults
bzip2                     1.0.6                h14c3975_5    defaults
ca-certificates           2019.5.15                     0    defaults
cairo                     1.14.12              h8948797_3    defaults
cdecimal                  2.3              py27h14c3975_3    defaults
certifi                   2019.3.9                 py27_0    defaults
cffi                      1.12.3           py27h2e261b9_0    defaults
chardet                   3.0.4                    py27_1    defaults
click                     7.0                      py27_0    defaults
cloudpickle               1.1.1                      py_0    defaults
clyent                    1.2.2                    py27_1    defaults
colorama                  0.4.1                    py27_0    defaults
conda                     4.6.14                   py27_0    defaults
conda-build               3.17.8                   py27_0    defaults
conda-env                 2.6.0                         1    defaults
conda-verify              3.1.1                    py27_0    defaults
configparser              3.7.4                    py27_0    defaults
contextlib2               0.5.5                    py27_0    defaults
cryptography              2.7              py27h1ba5d50_0    defaults
curl                      7.64.1               hbc83047_0    defaults
cycler                    0.10.0                   py27_0    defaults
cython                    0.29.10          py27he6710b0_0    defaults
cytoolz                   0.9.0.1          py27h14c3975_1    defaults
dask                      1.2.2                      py_0    defaults
dask-core                 1.2.2                      py_0    defaults
dbus                      1.13.6               h746ee38_0    defaults
decorator                 4.4.0                    py27_1    defaults
defusedxml                0.6.0                      py_0    defaults
distributed               1.28.1                   py27_0    defaults
docutils                  0.14                     py27_0    defaults
entrypoints               0.3                      py27_0    defaults
enum34                    1.1.6                    py27_1    defaults
et_xmlfile                1.0.1                    py27_0    defaults
expat                     2.2.6                he6710b0_0    defaults
fastcache                 1.1.0            py27h7b6447c_0    defaults
filelock                  3.0.10                   py27_0    defaults
flask                     1.0.3                    py27_0    defaults
fontconfig                2.13.0               h9420a91_0    defaults
freetype                  2.9.1                h8a8886c_1    defaults
fribidi                   1.0.5                h7b6447c_0    defaults
funcsigs                  1.0.2            py27h83f16ab_0    defaults
functools32               3.2.3.2                  py27_1    defaults
future                    0.17.1                   py27_0    defaults
futures                   3.2.0                    py27_0    defaults
get_terminal_size         1.0.0                haa9412d_0    defaults
gevent                    1.4.0            py27h7b6447c_0    defaults
glib                      2.56.2               hd408876_0    defaults
glob2                     0.6                      py27_1    defaults
gmp                       6.1.2                h6c8ec71_1    defaults
gmpy2                     2.0.8            py27h10f8cd9_2    defaults
graphite2                 1.3.13               h23475e2_0    defaults
greenlet                  0.4.15           py27h7b6447c_0    defaults
grin                      1.2.1                    py27_4    defaults
gst-plugins-base          1.14.0               hbbd80ab_1    defaults
gstreamer                 1.14.0               hb453b48_1    defaults
h5py                      2.9.0            py27h7918eee_0    defaults
harfbuzz                  1.8.8                hffaf4a1_0    defaults
hdf5                      1.10.4               hb1b8bf9_0    defaults
heapdict                  1.0.0                    py27_2    defaults
html5lib                  1.0.1                    py27_0    defaults
icu                       58.2                 h9c2bf20_1    defaults
idna                      2.8                      py27_0    defaults
imageio                   2.5.0                    py27_0    defaults
imagesize                 1.1.0                    py27_0    defaults
importlib_metadata        0.17                     py27_1    defaults
intel-openmp              2019.4                      243    defaults
ipaddress                 1.0.22                   py27_0    defaults
ipykernel                 4.10.0                   py27_0    defaults
ipython                   5.8.0                    py27_0    defaults
ipython_genutils          0.2.0                    py27_0    defaults
ipywidgets                7.4.2                    py27_0    defaults
isort                     4.3.20                   py27_0    defaults
itsdangerous              1.1.0                    py27_0    defaults
jbig                      2.1                  hdba287a_0    defaults
jdcal                     1.4.1                      py_0    defaults
jedi                      0.13.3                   py27_0    defaults
jinja2                    2.10.1                   py27_0    defaults
jpeg                      9b                   h024ee3a_2    defaults
jsonschema                3.0.1                    py27_0    defaults
jupyter                   1.0.0                    py27_7    defaults
jupyter_client            5.2.4                    py27_0    defaults
jupyter_console           5.2.0                    py27_1    defaults
jupyter_core              4.4.0                    py27_0    defaults
jupyterlab                0.33.11                  py27_0    defaults
jupyterlab_launcher       0.11.2           py27h28b3542_0    defaults
kiwisolver                1.1.0            py27he6710b0_0    defaults
krb5                      1.16.1               h173b8e3_7    defaults
lazy-object-proxy         1.4.1            py27h7b6447c_0    defaults
libarchive                3.3.3                h5d8350f_5    defaults
libcurl                   7.64.1               h20c2e04_0    defaults
libedit                   3.1.20181209         hc058e9b_0    defaults
libffi                    3.2.1                hd88cf55_4    defaults
libgcc-ng                 8.2.0                hdf63c60_1    defaults
libgfortran-ng            7.3.0                hdf63c60_0    defaults
liblief                   0.9.0                h7725739_2    defaults
libpng                    1.6.37               hbc83047_0    defaults
libsodium                 1.0.16               h1bed415_0    defaults
libssh2                   1.8.2                h1ba5d50_0    defaults
libstdcxx-ng              8.2.0                hdf63c60_1    defaults
libtiff                   4.0.10               h2733197_2    defaults
libtool                   2.4.6                h7b6447c_5    defaults
libuuid                   1.0.3                h1bed415_2    defaults
libxcb                    1.13                 h1bed415_1    defaults
libxml2                   2.9.9                he19cac6_0    defaults
libxslt                   1.1.33               h7d1a2b0_0    defaults
linecache2                1.0.0                    py27_0    defaults
llvmlite                  0.28.0           py27hd408876_0    defaults
locket                    0.2.0            py27h73929a2_1    defaults
lxml                      4.3.3            py27hefd8a0e_0    defaults
lz4-c                     1.8.1.2              h14c3975_0    defaults
lzo                       2.10                 h49e0be7_2    defaults
markupsafe                1.1.1            py27h7b6447c_0    defaults
matplotlib                2.2.3            py27hb69df0a_0    defaults
mccabe                    0.6.1                    py27_1    defaults
mistune                   0.8.4            py27h7b6447c_0    defaults
mkl                       2019.4                      243    defaults
mkl-service               2.0.2            py27h7b6447c_0    defaults
mkl_fft                   1.0.12           py27ha843d7b_0    defaults
mkl_random                1.0.2            py27hd81dba3_0    defaults
more-itertools            5.0.0                    py27_0    defaults
mpc                       1.1.0                h10f8cd9_1    defaults
mpfr                      4.0.1                hdf1c602_3    defaults
mpmath                    1.1.0                    py27_0    defaults
msgpack-python            0.6.1            py27hfd86e86_1    defaults
multipledispatch          0.6.0                    py27_0    defaults
navigator-updater         0.2.1                    py27_0    defaults
nbconvert                 5.5.0                      py_0    defaults
nbformat                  4.4.0            py27hed7f2b2_0    defaults
ncurses                   6.1                  he6710b0_1    defaults
networkx                  2.2                      py27_1    defaults
nltk                      3.4.1                    py27_0    defaults
nose                      1.3.7                    py27_2    defaults
notebook                  5.7.8                    py27_0    defaults
numba                     0.43.1           py27h962f231_0    defaults
numexpr                   2.6.9            py27h9e4a6bb_0    defaults
numpy                     1.16.4           py27h7e9f1db_0    defaults
numpy-base                1.16.4           py27hde5b4d6_0    defaults
numpydoc                  0.9.1                      py_0    defaults
olefile                   0.46                     py27_0    defaults
openpyxl                  2.6.2                      py_0    defaults
openssl                   1.1.1c               h7b6447c_1    defaults
packaging                 19.0                     py27_0    defaults
pandas                    0.24.2           py27he6710b0_0    defaults
pandoc                    2.2.3.2                       0    defaults
pandocfilters             1.4.2                    py27_1    defaults
pango                     1.42.4               h049681c_0    defaults
parso                     0.4.0                      py_0    defaults
partd                     0.3.10                   py27_1    defaults
patchelf                  0.9                  he6710b0_3    defaults
path.py                   11.5.0                   py27_0    defaults
pathlib2                  2.3.3                    py27_0    defaults
patsy                     0.5.1                    py27_0    defaults
pcre                      8.43                 he6710b0_0    defaults
pep8                      1.7.1                    py27_0    defaults
pexpect                   4.7.0                    py27_0    defaults
pickleshare               0.7.5                    py27_0    defaults
pillow                    6.0.0            py27h34e0f95_0    defaults
pip                       19.1.1                   py27_0    defaults
pixman                    0.38.0               h7b6447c_0    defaults
pkginfo                   1.5.0.1                  py27_0    defaults
pluggy                    0.12.0                     py_0    defaults
ply                       3.11                     py27_0    defaults
prometheus_client         0.6.0                    py27_0    defaults
prompt_toolkit            1.0.15                   py27_0    defaults
psutil                    5.6.2            py27h7b6447c_0    defaults
ptyprocess                0.6.0                    py27_0    defaults
py                        1.8.0                    py27_0    defaults
py-lief                   0.9.0            py27h7725739_2    defaults
pycairo                   1.18.1           py27h2a1e443_0    defaults
pycodestyle               2.5.0                    py27_0    defaults
pycosat                   0.6.3            py27h14c3975_0    defaults
pycparser                 2.19                     py27_0    defaults
pycrypto                  2.6.1            py27h14c3975_9    defaults
pycurl                    7.43.0.2         py27h1ba5d50_0    defaults
pyflakes                  2.1.1                    py27_0    defaults
pygments                  2.4.2                      py_0    defaults
pylint                    1.9.2                    py27_0    defaults
pyodbc                    4.0.26           py27he6710b0_0    defaults
pyopenssl                 19.0.0                   py27_0    defaults
pyparsing                 2.4.0                      py_0    defaults
pyqt                      5.9.2            py27h05f1152_2    defaults
pyrsistent                0.14.11          py27h7b6447c_0    defaults
pysocks                   1.7.0                    py27_0    defaults
pytables                  3.5.1            py27h71ec239_0    defaults
pytest                    4.6.2                    py27_0    defaults
python                    2.7.16               h9bab390_0    defaults
python-dateutil           2.8.0                    py27_0    defaults
python-libarchive-c       2.8                      py27_6    defaults
pytz                      2019.1                     py_0    defaults
pywavelets                1.0.3            py27hdd07704_1    defaults
pyyaml                    5.1              py27h7b6447c_0    defaults
pyzmq                     18.0.0           py27he6710b0_0    defaults
qt                        5.9.7                h5867ecd_1    defaults
qtawesome                 0.5.7                    py27_1    defaults
qtconsole                 4.5.1                      py_0    defaults
qtpy                      1.7.1                      py_0    defaults
readline                  7.0                  h7b6447c_5    defaults
requests                  2.22.0                   py27_0    defaults
rope                      0.14.0                     py_0    defaults
ruamel_yaml               0.15.46          py27h14c3975_0    defaults
scandir                   1.10.0           py27h7b6447c_0    defaults
scikit-image              0.14.2           py27he6710b0_0    defaults
scikit-learn              0.20.3           py27hd81dba3_0    defaults
scipy                     1.2.1            py27h7c811a0_0    defaults
seaborn                   0.9.0                    py27_0    defaults
send2trash                1.5.0                    py27_0    defaults
setuptools                41.0.1                   py27_0    defaults
simplegeneric             0.8.1                    py27_2    defaults
singledispatch            3.4.0.3          py27h9bcb476_0    defaults
sip                       4.19.8           py27hf484d3e_0    defaults
six                       1.12.0                   py27_0    defaults
snappy                    1.1.7                hbae5bb6_3    defaults
snowballstemmer           1.2.1            py27h44e2768_0    defaults
sortedcollections         1.1.2                    py27_0    defaults
sortedcontainers          2.1.0                    py27_0    defaults
soupsieve                 1.8                      py27_0    defaults
sphinx                    1.8.5                    py27_0    defaults
sphinxcontrib             1.0                      py27_1    defaults
sphinxcontrib-websupport  1.1.0                    py27_1    defaults
spyder                    3.3.4                    py27_0    defaults
spyder-kernels            0.4.4                    py27_0    defaults
sqlalchemy                1.3.4            py27h7b6447c_0    defaults
sqlite                    3.28.0               h7b6447c_0    defaults
ssl_match_hostname        3.7.0.1                  py27_0    defaults
statsmodels               0.9.0            py27h035aef0_0    defaults
subprocess32              3.5.4            py27h7b6447c_0    defaults
sympy                     1.4                      py27_0    defaults
tblib                     1.4.0                      py_0    defaults
terminado                 0.8.2                    py27_0    defaults
testpath                  0.4.2                    py27_0    defaults
tk                        8.6.8                hbc83047_0    defaults
toolz                     0.9.0                    py27_0    defaults
tornado                   5.1.1            py27h7b6447c_0    defaults
tqdm                      4.31.1                   py27_1    defaults
traceback2                1.4.0                    py27_0    defaults
traitlets                 4.3.2            py27hd6ce930_0    defaults
typing                    3.6.6                    py27_0    defaults
unicodecsv                0.14.1                   py27_0    defaults
unittest2                 1.1.0                    py27_0    defaults
unixodbc                  2.3.7                h14c3975_0    defaults
urllib3                   1.24.2                   py27_0    defaults
wcwidth                   0.1.7                    py27_0    defaults
webencodings              0.5.1                    py27_1    defaults
werkzeug                  0.15.4                     py_0    defaults
wheel                     0.33.4                   py27_0    defaults
widgetsnbextension        3.4.2                    py27_0    defaults
wrapt                     1.11.1           py27h7b6447c_0    defaults
wurlitzer                 1.0.2                    py27_0    defaults
xlrd                      1.2.0                    py27_0    defaults
xlsxwriter                1.1.8                      py_0    defaults
xlwt                      1.3.0                    py27_0    defaults
xz                        5.2.4                h14c3975_4    defaults
yaml                      0.1.7                had09818_2    defaults
zeromq                    4.3.1                he6710b0_3    defaults
zict                      0.1.4                    py27_0    defaults
zipp                      0.5.1                      py_0    defaults
zlib                      1.2.11               h7b6447c_3    defaults
zstd                      1.3.7                h0b5b093_0    defaults
```
</p></details>

**Comment by msarahan** [comment]
You are seeing this ping-pong because of noarch packages.  There's an explicit step in the solver in conda 4.7 to prevent this.  4.7.1 is available on the conda-canary channel.  It has issues with ping-ponging between channels if you're adding/dropping channels with -c, but it should otherwise work fine.  4.7.2 should be out on canary soon.

The fix will not be backported to the 4.6 series.

**Comment by msarahan** [comment]
4.7 is on defaults now. I think this is fixed. I'm closing this issue, but please reopen it open a new issue if you still see problems.
